documentation/doclava/android.gradle

```diff
@@ -26,57 +27,34 @@ task generateDoclava(type: Javadoc,
     doclet "com.google.doclava.Doclava"
     memberLevel = JavadocMemberLevel.PROTECTED
 
-    if(project.documentation.javadocOverview) {
+    if (project.documentation.javadocOverview) {
         overview = project.documentation.javadocOverview
     }
 
-    addStringOption "hdf project.name", "${project.name} - ${project.version}"
+    addStringOption "hdf project.name", "$project.name - $project.version"
     addStringOption "hdf project.version", project.version
-    addStringOption "federate android", "http://developer.android.com/reference" 
+    addStringOption "apiversion", project.version
+
+    addStringOption "federate android", "http://developer.android.com/reference"
     // TODO: generate more recent android platform xml file
     // see https://code.google.com/archive/p/doclava/wikis/FederatedDocs.wiki
-    addStringOption "federationxml android", "${CONFIG.configDir}/documentation/doclava/android-22.xml"
+    addStringOption "federationxml android", Paths.get("$CONFIG.configDir", "documentation", "doclava", "android-22.xml").toString()
     // TODO: generate doclava xml for support libs
     // TODO: customize template
-    // addStringOption "templatedir", "${CONFIG.configDir}/documentation/doclava/template"
+    // addStringOption "templatedir", Paths.get(CONFIG.configDir, "documentation", "doclava", "template").toString()
   }
 
-  destinationDir = new File(buildDir, 'javadocs')
-  classpath += project.files(android.getBootClasspath())
-  classpath += configurations.compile
-  classpath += configurations.provided
-  def aarDependencies = classpath.filter { it.name.endsWith('.aar') }
-  classpath -= aarDependencies
-  aarDependencies.each { aar ->
-    // Extract classes.jar from the AAR dependency, and add it to the javadoc classpath
-    def outputPath = "$buildDir/tmp/aarJar/${aar.name.replace('.aar', '.jar')}"
-    classpath += files(outputPath)
+  project.afterEvaluate {
+    destinationDir = new File(buildDir, 'javadocs')
 
-    // Use a task so the actual extraction only happens before the javadoc task is run
-    dependsOn task(name: "extract_${aar.name}").doLast {
-      extractEntry(aar, 'classes.jar', outputPath)
-    }
-  }
-}
+    // Android
+    classpath += files(android.getBootClasspath())
```

Comment:
i assume the `classpath` in the `Javadoc` task is not initialized as null? It is in the findbug task (so `+=` doesn't work the first time), so just double checking.

Reply:
Yes it's a valid non-null object